bugbear.py

```diff
@@ -1104,6 +1105,29 @@ def check_for_b022(self, node):
         ):
             self.errors.append(B022(node.lineno, node.col_offset))
 
+    def check_for_b908(self, node: ast.With):
+        has_multiline_body = len(node.body) > 1
+        for node_item in node.items:
+            if (
+                not hasattr(node_item.context_expr, "func")
```

Comment:
It looks so ugly, but I did not find a better solution
Do you have any suggestions to improve this block, let's say use some type check?

Reply:
First I would make a helper function like `is_assertRaises_like`, so you can make it `return False/True` with slightly simpler conditionals. Then the body of this can just be `if is_assertRaises_like(node_item.context_expr) and has_multiline_body:`. And instead of the hasattr checks, you can do `isinstance` checks on the appropriate AST classes, like `isinstance(expr, ast.Call)`.

Reply:
Thanks for the suggestion!